lib/parsing/ParsingUtils.ml

```diff
@@ -56,4 +63,46 @@ let str_to_col =
       ("brightcyan", BrightCyan);
       ("brightwhite", BrightWhite);
     ];
-  fun s -> Hashtbl.find_opt colors s
+  fun loc s ->
+    match Hashtbl.find_opt colors s with
+    | Some c -> c
+    | None ->
+        let pos = lexloc_to_pos loc in
+        let txt = Format.sprintf "Unknown color '%s'." s in
+        ErrorUtils.error txt pos
+
+let mk_imm loc i =
+  let pos = lexloc_to_pos loc in
+  match Immediate.of_int i with
+  | Some i -> mk_pos pos i
+  | None ->
+      let txt =
+        Format.sprintf "The value '%d' cannot be represented on 32 bit." i
```

Comment:
Un immédiat est pas censé tenir sur 16 et 24 bits ? Ce message d'erreur est peut être trop laxiste et un peu trompeur non ?

Ou alors cette fonction est-elle appelée que pour les immédiates dans la section .data ?

Reply:
On a plusieurs notions l'immédiat :
- Les constantes parsée dans le.richier source sont appelées des immédiats. Elles sont encodées sur 32bits
- Les arguments d'un Loadi (toujours un "immédiat" mais dans le programme c'est de type `UInt16.t`). Ils sont calculés à partir des constantes qui entrent en jeu dans les calculs (load d'adresse, load de constantes, etc)
- Les arguments d'un Jmpi (toujours un "immédiat" mais dans le programme c'est de type `Int24.t`). Même remarque qu'auparavant.

Le type `Immediate.t` encapsule à la fois les valeurs que peut prendre un `loadi r imm` et les valeurs déclarées dans un `.data`. Un load d'un mot de 32bit, est découpé en deux load sur chacune des parties du mot.

Reply:
C'est étrange d'appeler une constante dans .data un immédiat, mais pourquoi pas !
Sinon c'est possible de distinguer les deux ?

Reply:
Je renomme en `IntConstant` ?

Reply:
Fixed in 0316a489c1bef975a311e522ca3a34fd1dd9b04c.